### ISSUE TITLE
Firebase database limitToLast not working on Android

### ISSUE THREAD

**Issue by KevinTheGray** [flutter/flutter#10823]
## Steps to Reproduce
- Test on an Android device
- Use `firebase_database: "^0.0.7"`
- Make a query, something like 
```dart
Query query = itemsReference
        .orderByChild("published-date")
        .limitToLast(4);
```
- I get back ten items when I would expect 4

## Flutter Doctor
```
[✓] Flutter (on Mac OS X 10.12.4 16E195, locale en-US, channel master)
    • Flutter at /Users/KG/Developer/Flutter/flutter
    • Framework revision 6965312822 (2 days ago), 2017-06-16 10:05:18 -0700
    • Engine revision 784e975672
    • Tools Dart version 1.24.0-dev.6.7

[✓] Android toolchain - develop for Android devices (Android SDK 26.0.0)
    • Android SDK at /Users/KG/Library/Android/sdk
    • Platform android-26, build-tools 26.0.0
    • ANDROID_HOME = /Users/KG/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] iOS toolchain - develop for iOS devices (Xcode 8.3.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 8.3.3, Build version 8E3004b
    • ios-deploy 1.9.1
    • CocoaPods version 1.2.1

[✓] Android Studio (version 2.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Gradle version 3.2
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] IntelliJ IDEA Community Edition (version 2017.1.4)
    • Flutter plugin version 14.0
    • Dart plugin version 171.4694.29

[✓] Connected devices
    • Android SDK built for x86 • emulator-5554                            • android-x86 • Android 7.0 (API 24) (emulator)
    • Nexus 5X                  • 0106bcc49c8ca3eb                         • android-arm • Android 7.1.2 (API 25)
    • Kevin's iPhone            • c95c86bdc3aac565adb373de3dc2dbedcdecc900 • ios         • iOS 10.3.1 (14E304)
    • iPhone 5s                 • AB2FA880-541C-49D7-BDB2-9720261982A3     • ios         • iOS 10.3 (simulator)
```

**Comment by eseidelGoogle** [comment]
@collinjackson would know.

**Comment by collinjackson** [comment]
Investigating. Thanks.

**Comment by collinjackson** [comment]
I had meant to include this in #89, but didn't. It will be landed in #144.

**Comment by collinjackson** [comment]
Fixed in flutter/plugins#144 (version 0.0.8)

**Comment by kroikie** [comment]
@KevinTheGray

This issue has been moved to https://github.com/FirebaseExtended/flutterfire/issues/993. Any further collaboration will be done there.
